mysql/toddl.go

```diff
@@ -136,10 +136,10 @@ func toSpannerType(conv *internal.Conv, id string, mods []int64) (ddl.ScalarType
 		}
 		return ddl.String{Len: ddl.MaxLength{}}, nil
 	case "text", "tinytext", "mediumtext", "longtext":
-		maxExpectedMods(0)
+		maxExpectedMods(1)
```

Comment:
Can you give some background on this change: can text types have a modifier?
Can you give an example?

Reply:
Though we would not be specifying mods for `text` type, but MySQL by default will put max value for `text` type in `CHARACTER_MAXIMUM_LENGTH` column of `information_schema.COLUMNS` table. Eg: 
```
mysql> select c.column_name,c.data_type,c.is_nullable,c.column_default, c.character_maximum_length, c.numeric_precision, c.numeric_scale from information_schema.COLUMNS c where table_schema='testdb' and COLUMN_NAME LIKE '%text';
+-------------+------------+-------------+----------------+--------------------------+-------------------+---------------+
| COLUMN_NAME | DATA_TYPE  | IS_NULLABLE | COLUMN_DEFAULT | CHARACTER_MAXIMUM_LENGTH | NUMERIC_PRECISION | NUMERIC_SCALE |
+-------------+------------+-------------+----------------+--------------------------+-------------------+---------------+
| text        | text       | YES         | NULL           |                    65535 |              NULL |          NULL |
| tinytext    | tinytext   | YES         | NULL           |                      255 |              NULL |          NULL |
| mediumtext  | mediumtext | YES         | NULL           |                 16777215 |              NULL |          NULL |
| longtext    | longtext   | YES         | NULL           |               4294967295 |              NULL |          NULL |
+-------------+------------+-------------+----------------+--------------------------+-------------------+---------------+
4 rows in set (0.00 sec)
```

In function `toType()`, when case is `charLen.Valid` we return `schema.Type{Name: dataType, Mods: []int64{charLen.Int64}}`.  Thus, for `text` type, mods will be allocated, and hence I have changed to`maxExpectedMods(1)` in `mysql/toddl.go` file.

Reply:
Thanks for all of the details and the example.

So this doesn't show up in mysql dump parsing, just in the info schema tables?

In effect we're relaxing the error checking for mysql dump generated schemas, but I don't see a huge issue in doing that.

Reply:
Correct, In` mysqldump` create statements will have `text` and `blob` datatype without any mods. It's also mentioned in pingcap documentation under data types section ([LINK](https://docs.pingcap.com/tidb/dev/sql-statement-create-table)).
Even if mysqldump has a modifier for `text` & `blob` datatype then also pingcap parser will ignore it and return the parsed statement without any modifier ( I have tested the same for both datatypes). Hence, we are not relaxing error checking for mysqldump. 
One work around could be to explicitly put a condition when datatype is `text` or `blob` and return `schema.Type{Name: dataType}` in mysql/infoschema.go  and then we do not need to change `maxExpectedMods(0)` but I do not see much benefit of doing that and also it would not be a good code design.

Thus, I think we can move forward with this change of error checks.